.buildkite/pipeline.yml

```diff
@@ -2,8 +2,8 @@
 common_params:
   # Common plugin settings to use with the `plugins` key.
   - &common_plugins
-    - automattic/bash-cache#v1.0.0: ~
-    - automattic/git-s3-cache#add/vm-host-support:
+    - automattic/bash-cache#add/cocoapods-support: ~
+    - automattic/git-s3-cache#v1.1.0:
```

Comment:
Lazy question: How come did this change? What's the role of `~` after the `:`?

Reply:
It's from the Buildkite docs – tbh I have no idea what it does 😜

Reply:
`~` is one of the possible ways in YAML to represents a `null` value.
 - Another option being just to not specify the value at all (remove the `~` and keep that empty) but is generally not favored as it could lead to ambiguity (especially making you wonder if the next line is a separate key or the nested value of the current key)
 - A more common option is the English spelling `null`. Some prefer `null` for its explicitness, other argue that `~` help avoid the risk of some people thinking `null` could mean the `"null"` literal string
 - The last option is to force the type using a tag (tags in YAML start with `!!` and provide metadata info to the YAML parser, like the type in this case): `!!null`

Here, since the `bash-cache` plugin doesn't need additional parameters (unlike other plugins like the `docker` or `git-s3-cache` plugins typically do), the value associated with that key is null (as opposed to a nested dictionary providing the additional parameters). So here with `~` it's a bit like you're basically passing an empty dictionary for the parameters of `bash-cache` (but not sure we can create empty dictionaries in YAML, but we can pass `null` which is what `~` is here)

Reply:
As an additional note, the way CircleCI handles those cases (of things sometimes having extra parameters, sometimes not) is to accept base strings instead of a key with null value. For example in CircleCI we'd have:

```yml
Screenshots:
    when: << pipeline.parameters.generate_screenshots >>
    jobs:
      - raw-screenshots
      - promo-screenshots:
          requires: [raw-screenshots]
```
Where `jobs` is an array for which each item could be either a single String (`raw-screenshots`) with the name of a job… or a single-item dictionary (`promo-screenshots: <extra options here>`) depending if you need to specify extra options or not. _(Bonus: note how here in the extra options passed to `promo-screenshots`, the `requires:` option uses `[raw-screenshots]` which is an array, but declared with `[]` brackets (JSON-style) and not with the `-` one-item-per-line-dash-list syntax (more common YAML-style) for conciseness. Both are valid, because YAML is a superset of JSON.)_

While in BuildKite syntax, it seems that in those kind of lists, BuildKite expects every item to be a single-item dictionary (rather than allowing sometimes a String, sometimes a single-item dict like CircleCI). And then for items that don't need extra options, expect you to use null (`~`) as the value for those non-existing options.

I have no idea if BuildKite syntax is lenient enough to also allow you to use a single String instead of a "single-item dictionary with null value" in those cases and if both are valid in BuildKite or not, but if the BuildKite docs suggests to use `~`, I think it's best to use what they suggest :wink: